apps/forms/custom.py

```diff
@@ -40,6 +56,17 @@ def _setup_form_helper(self):
             ),
             SRVCommonDivField("port", placeholder="8000"),
             SRVCommonDivField("image"),
+            Accordion(
+                AccordionGroup(
+                    "Advanced settings",
+                    PrependedText(
+                        "default_url_subpath",
+                        "Subdomain/",
+                        template="apps/partials/srv_prepend_input_group_custom_app.html",
```

Comment:
```suggestion
                        mark_safe("<span id='id_custom_default_url_prepend'>Subdomain/</span>"),
                        template="apps/partials/srv_prepend_append_input_group.html",
```

I think one way to this this is to not use a separate template but use `apps/partials/srv_prepend_input_group_custom_app.html`, add an id to the Prepend input group element using span so that it is accessible through the javascript id match that you use. I have also tried to shorten the id, you can ofcourse choose another name.

Then I suggest moving parts of your javascript to the `templates/apps/create_base.html` file. The changes can be the following:

Load get_setting
```python
{% load get_setting %}
```

In the script section make the following changes
```diff
+ const subdomainInput = document.getElementById("id_subdomain");
+ const customDefaultUrlFormModifiedUrlText = document.getElementById("id_custom_default_url_prepend");

+ function updateUrlWithSubdomain() {
+        const subdomainValue = subdomainInput.value.trim();
+        const displayValue = subdomainValue || "subdomain_name";
+      customDefaultUrlFormModifiedUrlText.innerHTML = `${displayValue}.{% get_setting 'DOMAIN' %}/</b>`;
+    }

    // User is "finished typing," do something
    function doneTyping () {


        let subdomain_input = $("#id_subdomain").val();
        if (subdomain_input.length == 0) {
            //console.log("Empty subdomain input");
            clearSubdomainValidation();
            // Empty text field. Enable the submit button and clear any validation styles.
            $("#submit-id-submit").prop('disabled', false);
        }
        else {
            //console.log(`Checking subdomain validity for: ${subdomain_input}`);
            // Run check if there is text in the input field
            // Convert the input subdomain to lowercase for the validation.
            // OK because the input form and the server does the same.
            checkSubdomainValidity(subdomain_input.toLowerCase());
        }
+        updateUrlWithSubdomain();
    }
........
........
........
 htmx.onLoad(function(content) {
        runScript();
+       updateUrlWithSubdomain();
    })
```

We already have an eventlistener listening to keystrokes on the subdomain input box. So we can use that one to update the URL for your input prepend text.

Reply:
Thanks a lot for this suggestion. It helped me a lot to understand the existing scripts.